extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -438,7 +465,21 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 
 						this._installCompletion.resolve();
 						this._installInProgress = false;
-						await vscode.commands.executeCommand('notebook.action.restartJupyterNotebookSessions');
+						if (this._upgradeInProcess) {
+							// Pass in false for restartJupyterServer parameter since the jupyter server has already been shutdown
+							// when removing the old Python version on Windows.
+							if (process.platform === constants.winPlatform) {
+								await vscode.commands.executeCommand('notebook.action.restartJupyterNotebookSessions', false);
```

Comment:
Why is the logic different for Windows?

Reply:
Windows does not allow us to delete the Python files when it is in use. So we first have to shut down active notebook sessions before trying to remove the old python installation. On mac/linux we can delete the old python installation even if there are python notebook sessions running. Once the installation is complete, the open notebook sessions are restarted with the new python installation.